src/test/java/core/basesyntax/impl/FruitTransactionParserImplTest.java

```diff
@@ -0,0 +1,96 @@
+package core.basesyntax.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionParserImplTest {
+    private static FruitTransactionParserImpl parser;
+    private List<String> transactionStrings;
+
+    @BeforeClass
+    public static void beforeClass() {
+        parser = new FruitTransactionParserImpl();
+    }
+
+    @Before
+    public void setUp() {
+        transactionStrings = new ArrayList<>();
+        transactionStrings.add("type,fruit,quantity");
+        transactionStrings.add("b,banana,18");
+        transactionStrings.add("s,peach,35");
+        transactionStrings.add("r,apple,6");
+        transactionStrings.add("p,mango,7");
+    }
+
+    @Test
+    public void parse_validList_Ok() {
+        List<FruitTransaction> actual = parser.parse(transactionStrings);
+        for (FruitTransaction fruitTransaction: actual) {
+            System.out.println(fruitTransaction.getFruit());
+            System.out.println(fruitTransaction.getAmount());
+            System.out.println(fruitTransaction.getOperation());
+        }
+        List<FruitTransaction> expected = getListOfFruitTransactions();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void parse_emptyList_Ok() {
+        List<FruitTransaction> actual = parser.parse(new ArrayList<>());
+        List<FruitTransaction> expected = List.of();
+        assertEquals(expected, actual);
+    }
+
+    @Test (expected = RuntimeException.class)
```

Comment:
you haven’t corresponding checks in your solution. So now you are testing system exceptions. Add checking  to your solution or remove this test case (check  it everywhere)